server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1038,7 +1039,7 @@ public void cleanup(long repositoryStateId, Version repositoryMetaVersion, Actio
                     logger.info("[{}] Found stale root level blobs {}. Cleaning them up", metadata.name(), blobsToLog);
                 }
             }
-            deleteFromContainer(blobContainer(), blobsToDelete);
+            deleteFromContainer(blobContainer(), blobsToDelete.iterator());
```

Comment:
We can return an iterator in `BlobStoreRepository#staleRootBlobs` as we do in `#resolveFilesToDelete`, right?

Reply:
We could but I really like the info level logging of these. It's been very helpful in past debugging and would be somewhat awkward without a collection here. This is always going to be very few blobs anyway so I'd say it's not worth the extra complexity?